oracle/src/events/processAMBAffirmationRequests/index.js

```diff
@@ -22,6 +29,45 @@ function processAffirmationRequestsBuilder(config) {
       validatorContract = await getValidatorContract(bridgeContract, web3)
     }
 
+    // process retry queue
+    const retryQueue = await getRetryQueue('amb')
+    if (retryQueue.length > 0) {
+      rootLogger.info(`Processing ${retryQueue.length} transaction from retry queue`)
+      for (const queueItem of retryQueue) {
+        let gasEstimate
+        const { transactionHash, messageId, message } = queueItem
+        const logger = rootLogger.child({
+          eventTransactionHash: transactionHash
+        })
+
+        logger.info(`Processing affirmationRequest ${transactionHash}, messageId ${messageId} in retryQueue`)
+
+        try {
+          gasEstimate = await estimateGas({
+            web3,
+            homeBridge: bridgeContract,
+            validatorContract,
+            message,
+            address: config.validatorAddress,
+            transactionHash,
+            messageId
+          })
+          logger.debug({ gasEstimate }, 'Gas estimated')
+          await deleteFromRetryList(JSON.stringify({ bridge: 'amb', transactionHash, messageId, message }))
+        } catch (e) {
+          logger.error(e)
+        }
+
+        const data = bridgeContract.methods.executeAffirmation(message).encodeABI()
```

Comment:
Why does `executeAffirmation` is called even if `estimateGas` throws an error?

Reply:
`executeAffirmation` will not be called if error is thrown. If a tx not approved by Hashi, it will be retried in retryQueue. [This line](https://github.com/gnosischain/tokenbridge/blob/6852ab37504fdc1eb646cbc0827a522975b01d7a/oracle/src/events/processAMBAffirmationRequests/index.js#L131) will filter out duplicated tx if there is any.

Reply:
When a new event is detected, the event from retryQueue will be processed first, then the normal event.

Reply:
ok got it!